crates/primitives/src/lib.rs

```diff
@@ -203,12 +203,10 @@ pub trait TxTrace {
 
     /// Try to build a typed transaction
     fn try_build_typed_tx(&self) -> Result<TypedTransaction, SignatureError> {
-        let chain_id = self.chain_id();
-
         let tx = match self.ty() {
             0x0 => {
                 let tx = TxLegacy {
```

Comment:
whats the reason for the change, it is causing EF testcases to fail `transaction validation error` @lightsing